controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -372,9 +396,22 @@ public void getDelegationToken(io.pravega.controller.stream.api.grpc.v1.Controll
                     });
         } catch (Exception e) {
             log.error("Controller api failed with authenticator error");
+            logAndUntrackRequestTag(requestTag);
```

Comment:
I would put this in a `finally` block - everywhere.
Similarly with futures.

Reply:
Done.